File: C++/Program/DPR.cpp
/*
Saya Shidiq Arifin Sudrajat [2202152] mengerjakan soal LP1 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin
*/
#include <iostream>
#include <string>

using namespace std;

class DPR//Class DPR
{   //Attribute private
    private:
        string id;//Id anggota DPR
        string name;//Nama anggota DPR
        string name_bidang;//Nama bidang anggota DPR
        string name_partai;//Nama partai anggota DPR
    //Method public
    public:
        DPR()//Constructor
        {   //Set default
            this->id = "";
            this->name = "";
            this->name_bidang = "";
            this->name_partai = "";
        }

        DPR(string id, string name, string name_bidang, string name_partai)//Constructor dengan parameter
        {   //Set sesuai dengan paramete
            this->id = id;
            this->name = name;
            this->name_bidang = name_bidang;
            this->name_partai = name_partai;
        }

        string get_id()//Get id
        {
            return this->id;
        }

        void set_id(string id)//Set id
        {
            this->id = id;
        }
        
        string get_name()//Get nama
        {
            return this->name;
        }

        void set_name(string name)//Get nama
        {
            this->name = name;
        }

        string get_name_bidang()//Get nama bidang
        {
            return this->name_bidang;
        }

        void set_name_bidang(string name_bidang)//Set nama bidang
        {
            this->name_bidang = name_bidang;
        }

        string get_name_partai()//get nama partai
        {
            return this->name_partai;
        }

        void set_name_partai(string name_partai)//Set nama partai
        {
            this->name_partai = name_partai;
        }
        //Destructor
        ~DPR()
        {
            
        }
};

File: Java/Program/DPR.java
/*
Saya Shidiq Arifin Sudrajat [2202152] mengerjakan soal LP1 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin
*/
class DPR//Class DPR
{   //Private Attributes
    private String id;//Id
    private String name;//Nama
    private String nameBidang;//Nama Bidang
    private String namePartai;//Nama Partai
    //Method Public
    public DPR()//Constructor
    {
        this.id = "";
        this.name = "";
        this.nameBidang = "";
        this.namePartai = "";
    }

    public DPR(String id, String name, String nameBidang, String namePartai)//Constructor with parameter
    {
        this.id = id;
        this.name = name;
        this.nameBidang = nameBidang;
        this.namePartai = namePartai;
    }
    //Getter-Setter
    public String getId()
    {
        return id;
    }

    public void setId(String id)
    {
        this.id = id;
    }

    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    public String getNameBidang()
    {
        return nameBidang;
    }

    public void setNameBidang(String nameBidang)
    {
        this.nameBidang = nameBidang;
    }

    public String getNamePartai()
    {
        return namePartai;
    }

    public void setNamePartai(String namePartai)
    {
        this.namePartai = namePartai;
    }
}